core/src/main/java/org/fao/geonet/web/LocaleRedirects.java

```diff
@@ -156,7 +160,13 @@ private String createServiceUrl(HttpServletRequest request, String service, Stri
                 queryString = "?" + headers;
             }
         }
-        return request.getContextPath() + "/" + node + "/" + lang + "/" + service + queryString;
+        SettingManager sm = ApplicationContextHolder.get().getBean(SettingManager.class);
+        if(sm!=null && sm.getValue(Settings.SYSTEM_SERVER_HOST)!=null) {
```

Comment:
It is also possible to define a variable for 'request.getContextPath() + "/" + node + "/" + lang + "/" + service + queryString ' and/or to use a String.format, and so being more explicit with what this path part is.

Furthermore, I think sm can't be null, and one can also define a sm.getValue which take a default value, used if not found, which can be in this case '""'.